source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1109,21 +1109,55 @@ updateRSFromPrimary
         return
 
     # Election ids are ObjectIds, see
-    # "using setVersion and electionId to detect stale primaries"
+    # "Using electionId and setVersion to detect stale primaries"
     # for comparison rules.
 
-    # Null values for both electionId and setVersion are always considered less than
-    if serverDescription.electionId > topologyDescription.maxElectionId or (
-        serverDescription.electionId == topologyDescription.maxElectionId
-        and serverDescription.setVersion >= topologyDescription.maxSetVersion
-    ):
-        topologyDescription.maxElectionId = serverDescription.electionId
-        topologyDescription.maxSetVersion = serverDescription.setVersion
+    if serverDescription.maxWireVersion >= 17:  # MongoDB 6.0+
+        # Null values for both electionId and setVersion are always considered less than
+        if serverDescription.electionId > topologyDescription.maxElectionId or (
+            serverDescription.electionId == topologyDescription.maxElectionId
+            and serverDescription.setVersion >= topologyDescription.maxSetVersion
+        ):
+            topologyDescription.maxElectionId = serverDescription.electionId
+            topologyDescription.maxSetVersion = serverDescription.setVersion
+        else:
+            # Stale primary.
+            # replace serverDescription with a default ServerDescription of type "Unknown"
+            checkIfHasPrimary()
+            return
     else:
-        # Stale primary.
-        # replace serverDescription with a default ServerDescription of type "Unknown"
-        checkIfHasPrimary()
-        return
+        # Maintain old comparison rules, namely setVersion is checked before electionId
+        # see "Using electionId and setVersion to detect stale primaries"
+        # for comparison rules.
+
+        if serverDescription.setVersion is not null and serverDescription.electionId is not null:
+            # Election ids are ObjectIds, see
+            # "using setVersion and electionId to detect stale primaries"
+            # for comparison rules.
```

Comment:
This comment is a duplicate, can we remove it?

Reply:
done